src/util/domains/purchase-domain-if-available.ts

```diff
@@ -62,7 +62,8 @@ export default async function purchaseDomainIfAvailable(
       result instanceof ERRORS.DomainNotAvailable ||
```

Comment:
@juliangruber I think this pattern matching is for when we used to have Flow and it wasn't smart enough to infer the types when checking against `Error`. Since all of these are inheriting from the `Error` prototype I'm quite sure we can just remove this lines into a simpler:

```ts
if (result instanceof Error) {
  return result
}
```

Please try it out as it makes the code easier. Same for other places where you needed to add the check.

Reply:
Oh wow, I was wondering about all those manual checks. Will refactor!

Reply:
https://github.com/zeit/now-cli/pull/1953

Reply:
Let's keep refactoring this in that PR and merge this following the current structure